src/encoder.rs

```diff
@@ -99,29 +111,159 @@ impl From<FormatErrorKind> for FormatError {
+        let mut data = [0; 8];
+        data[..4].copy_from_slice(&self.num_frames.to_be_bytes());
+        data[4..].copy_from_slice(&self.num_plays.to_be_bytes());
+        chunk::encode_chunk(w, chunk::acTL, &data)
+    }
+}
+
+impl ScaledFloat {
+    pub fn encode<W: Write>(self, w: &mut W) -> io::Result<()> {
+        chunk::encode_chunk(w, chunk::gAMA, &self.into_scaled().to_be_bytes())
+    }
+}
+
+impl SourceChromaticities {
+    #[rustfmt::skip]
+    pub fn to_be_bytes(self) -> [u8; 32] {
+        let white_x = self.white.0.into_scaled().to_be_bytes();
+        let white_y = self.white.1.into_scaled().to_be_bytes();
+        let red_x   = self.red.0.into_scaled().to_be_bytes();
+        let red_y   = self.red.1.into_scaled().to_be_bytes();
+        let green_x = self.green.0.into_scaled().to_be_bytes();
+        let green_y = self.green.1.into_scaled().to_be_bytes();
+        let blue_x  = self.blue.0.into_scaled().to_be_bytes();
+        let blue_y  = self.blue.1.into_scaled().to_be_bytes();
+        [
+            white_x[0], white_x[1], white_x[2], white_x[3],
+            white_y[0], white_y[1], white_y[2], white_y[3],
+            red_x[0],   red_x[1],   red_x[2],   red_x[3],
+            red_y[0],   red_y[1],   red_y[2],   red_y[3],
+            green_x[0], green_x[1], green_x[2], green_x[3],
+            green_y[0], green_y[1], green_y[2], green_y[3],
+            blue_x[0],  blue_x[1],  blue_x[2],  blue_x[3],
+            blue_y[0],  blue_y[1],  blue_y[2],  blue_y[3],
+        ]
+    }
+
+    pub fn encode<W: Write>(self, w: &mut W) -> io::Result<()> {
+        chunk::encode_chunk(w, chunk::cHRM, &self.to_be_bytes())
+    }
+}
+
+impl SrgbRenderingIntent {
+    pub fn encode<W: Write>(self, w: &mut W) -> io::Result<()> {
+        chunk::encode_chunk(w, chunk::sRGB, &[self.into_raw()])
+    }
+}
+
+impl Info {
+    pub fn encode<W: Write>(&self, w: &mut W) -> io::Result<()> {
+        chunk::IHDR_encode(
+            w,
+            self.width,
+            self.height,
+            self.bit_depth,
+            self.color_type,
+            self.interlaced,
+        )?;
+
+        if let Some(p) = &self.palette {
+            chunk::encode_chunk(w, chunk::PLTE, p)?;
+        };
+
+        if let Some(t) = &self.trns {
+            chunk::encode_chunk(w, chunk::tRNS, t)?;
+        }
+
+        // If specified, the sRGB information overrides the source gamma and chromaticities.
+        if let Some(srgb) = &self.srgb {
+            let gamma = crate::srgb::substitute_gamma();
+            let chromaticities = crate::srgb::substitute_chromaticities();
+            srgb.encode(w)?;
+            gamma.encode(w)?;
+            chromaticities.encode(w)?;
+        } else {
+            self.source_gamma.map_or(Ok(()), |v| v.encode(w))?;
+            self.source_chromaticities.map_or(Ok(()), |v| v.encode(w))?;
+        }
+        self.animation_control.map_or(Ok(()), |v| v.encode(w))?;
```

Comment:
`map_or` is _clever_, it is not particularly readable and not necessarly. `clippy` will tell you to use `if let` instead. I'm puzzled why that was changed in the first place.